File: f77/build07/fcomp.f
C
        Subroutine Flight_Computer_Data
C
C       This subroutine sets the default data for the flight computer.
C
C       Inputs:
C           None.
C
C       Outputs:
C           Guidance_Gain - Commanded angular rate per measured LOS rate [Real]
C
C       Internal variables and constants:
C           None.
C
C       Declare global common and assign variable locations
C
        Include '../driver/global.inc'
        Include '../driver/sysvars.inc'
        Equivalence (Real_Array(301), Guidance_Gain)
C
C       Begin default data definitions:
C
        Guidance_Gain = 3.0e0
C
        Return
        End

        Subroutine Flight_Computer_Init
C
C       This subroutine initializes the variables for the flight computer.
C
C       Inputs:
C           None.
C       Outputs:
C           None.
C
C       Internal variables and constants:
C           None.
C
C       Begin math model initialization:
C
C       No initialization required.
C
        Return
        End

        Subroutine Flight_Computer
C
C       This subroutine determines the commanded airframe rate.
C
C       Inputs:
C           Guidance_Gain - Commanded angular rate per measured LOS rate (Real)
C           Q_s_Meas - Measured LOS rate [rad/sec]
C
C       Outputs:
C           ThetaDot_b_Cmd - Commanded attitude rate of missile [rad/sec]
C
C       Internal variables and constants:
C           None.
C
C       Declare global common and assign variable locations
C
        Include '../driver/global.inc'
        Include '../driver/sysvars.inc'
        Equivalence (Real_Array(300), ThetaDot_b_Cmd)
        Equivalence (Real_Array(301), Guidance_Gain)
        Equivalence (Real_Array(601), Q_s_Meas)
C
C       Begin math model:
C
C       Calculate guidance command
C
        ThetaDot_b_Cmd = Guidance_Gain*Q_s_Meas
C
        Return
        End

File: compcert/build08/fcomp.c
#include "fcomp.h"
#include "../driver/global.h"
#include "../driver/sysvars.h"

#define guidance_gain  real_array[301].r // Commanded angular rate per measured LOS rate [Real]
#define q_b_cmd        real_array[300].r //  Comanded attitude rate of missile [rad/sec]
#define q_s_meas       real_array[601].r // Measured LOS rate [rad/sec]

void flight_computer_data() {
  // This subroutine sets the default data for the flight computer.
  //
  // Inputs:
  //     None.
  //
  // Outputs:
  //     Guidance_Gain - Commanded angular rate per measured LOS rate [Real]
  //  Internal variables and constants:
  //      None.
  //
  //  Declare global common and assign variable locations

  // Begin default data definitions:
  guidance_gain = 3.0;
}

void flight_computer_init() {
  // This subroutine initializes the variables for the flight computer.
  //
  // Inputs:
  //     None.
  // Outputs:
  //     None.
   // Internal variables and constants:
  //     None.
  //
  // Begin math model initialization:
  //
  // No initialization required.
}

void flight_computer(void) {
  // This subroutine determines the commanded airframe rate.
  // Inputs:
  //     Guidance_Gain - Commanded angular rate per measured LOS rate [Real]
  //     Q_s_Meas - Measured LOS rate [rad/sec]
  // Outputs:
  //     Q_b_Cmd - Comanded attitude rate of missile [rad/sec]
  // Internal variables and constants:
  //     None.
  // Begin math model:
  // Calculate guidance command
  q_b_cmd = guidance_gain*q_s_meas;
}
